datafusion/src/physical_plan/hash_join.rs

```diff
@@ -1590,36 +1591,36 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn join_outer_one() -> Result<()> {
+    async fn join_full_one() -> Result<()> {
         let left = build_table(
             ("a1", &vec![1, 2, 3]),
             ("b1", &vec![4, 5, 7]), // 7 does not exist on the right
             ("c1", &vec![7, 8, 9]),
         );
         let right = build_table(
             ("a2", &vec![10, 20, 30]),
-            ("b1", &vec![4, 5, 6]),
+            ("b2", &vec![4, 5, 6]),
             ("c2", &vec![70, 80, 90]),
         );
-        let on = &[("b1", "b1")];
+        let on = &[("b1", "b2")];
 
         let join = join(left, right, on, &JoinType::Full)?;
 
         let columns = columns(&join.schema());
-        assert_eq!(columns, vec!["a1", "b1", "c1", "a2", "c2"]);
+        assert_eq!(columns, vec!["a1", "b1", "c1", "a2", "b2", "c2"]);
 
         let stream = join.execute(0).await?;
         let batches = common::collect(stream).await?;
 
         let expected = vec![
-            "+----+----+----+----+----+",
-            "| a1 | b1 | c1 | a2 | c2 |",
-            "+----+----+----+----+----+",
-            "|    |    |    | 30 | 90 |",
-            "| 1  | 4  | 7  | 10 | 70 |",
-            "| 2  | 5  | 8  | 20 | 80 |",
-            "| 3  | 7  | 9  |    |    |",
-            "+----+----+----+----+----+",
+            "+----+----+----+----+----+----+",
```

Comment:
I see -- the output was only showing `b1` from `left` -- that was very confusing. Thank you for making it clearer